components/Dashboard/Cards/SpaceCardModal.js

```diff
@@ -86,16 +88,20 @@ const SpaceCardModal = ({ handleClose, open, children, friends, participants, ho
   const router = useRouter();
   const [client, setClient] = useRecoilState(clientState.client);
   const [roomIsLoading, setRoomIsLoading] = useState(false);
-
-  // const gqlClient = useApolloClient();
-  // const { loading, error, data } = useQuery(GET_SESSION_USER);
-  // console.debug('data', data);
+  const [addUserToSpace] = useMutation(ADD_USER_TO_SPACE);
 
   const joinSpace = () => {
     // Add client to participant list
     setRoomIsLoading(true);
+    const addUserToSpaceInput = {
+      userId: client?.user?._id,
+      spaceId,
+    };
+    addUserToSpace({ variables: { addUserToSpaceInput } });
     router.push(`/room/${spaceId}`);
+    setRoomIsLoading(false);
```

Comment:
you can remove this line since the user is gonna be redirected anyway. Adding a setState in a function that executes after a redirect will lead to a memory leak (it should also say so in your console).